.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,33 @@
+name: GitHub Pages
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main  # Set a branch name to trigger deployment
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  deploy:
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Build
+        run: nox -s docs
```

Comment:
should: Do we not need to install `python`/`nox` on the workflow first?

Reply:
I agree, I think there should be a simple `setup-python@v5` step there.
In a step after that, we'll also need to install the requirements, right?